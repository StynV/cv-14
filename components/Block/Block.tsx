"use client"
import { ReactNode } from 'react'
import { useInView } from 'react-intersection-observer'
import YGWYS from '@/components/YGWYS/YGWYS'
import styles from './block.module.scss'

const Block = ({ title, period, children }: { title: string, period?: string, children: ReactNode }) => {
  const { ref, inView } = useInView({
    triggerOnce: true,
    threshold: 0.5, // Adjust as needed
  })
  
  return (
  <article
    className={`${styles.block} ${inView ? styles.inView : styles.notInView}`}
    ref={ref}
    itemScope
    itemType='https://schema.org/Article'
  >
    <>
      <meta itemProp='datePublished' content='2023-10-15' />
      <meta itemProp='dateModified' content='2024-02-03' />
      <meta itemProp='publisher' content='https://www.linkedin.com/in/styn-vercauteren-96918018b/' />
      
      <div className={styles.squareBlock}>
        <div className={styles.square}></div>
        <div className={styles.line}>
          <div className={styles.squareLeft}></div>
          <div className={styles.squareRight}></div>
        </div>
      </div>

      <div className={styles.textBlock}>
        <div className={styles.blockHeader}>
          <YGWYS html={title} className={styles.title} />
          {period && <h3 className={styles.period}>{period}</h3>}
        </div>
        {children}
      </div>
    </>
  </article>
  )
}

export default Block