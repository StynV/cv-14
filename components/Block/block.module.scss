@import "../../styles/vars.scss";

.block {
  @extend %page-container;

  grid-template-areas: 'square square textBlock textBlock textBlock textBlock textBlock textBlock textBlock textBlock textBlock textBlock';
  
  @media (max-width: $break-lg) {
    grid-template-areas: 'square square textBlock textBlock textBlock textBlock textBlock textBlock textBlock textBlock textBlock textBlock';
  }
  
  gap: calc(#{$spacing-bace} * 1);

  &.inView {
    animation: slideIn 0.5s forwards;
  }

  &.notInView {
    opacity: 0;
  }

  .squareBlock {
    grid-area: square;

    display: flex;
    flex-direction: column;
    align-items: center;
    width: 100%;
    
    .square {
      min-width: 4rem;
      min-height: 4rem;

      @media (max-width: $break-lg) {
        min-width: 2rem;
        min-height: 2rem;
      }

      background-color: #E3FDFD;
      transform: rotate(45deg);
    }

    .line {
      height: 100%;
      
      .squareRight {
        border-left: 1px solid lightgray;
        height: 100%;
      }
    }

  }

  .textBlock {
    grid-area: textBlock;
    
    @media (min-width: $break-lg) {
      margin-top: calc(#{$spacing-bace} * 1);
    }
    
    .blockHeader {
      display: flex;
      margin-bottom: calc(#{$spacing-bace} * 1);

      @media (min-width: $break-lg) {
        align-items: center;
      }

      @media (max-width: $break-lg) {
        flex-direction: column;
      }

      .title {
        flex-grow: 1;
      }

      .period {
        font-weight: 100;
      }
    }
  }
}