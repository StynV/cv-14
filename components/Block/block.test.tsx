import { render, screen } from '@testing-library/react'
import Block from './Block'

describe('Block', () => {
  it('renders a Block', () => {
    const { container, getByText } = render(<Block title='title' period='period'><p>children</p></Block>)

    expect(container).toMatchSnapshot()

    expect(container.querySelector('article')).toHaveClass('block notInView')
    expect(container.getElementsByClassName('squareBlock')[0]).toBeVisible()
    expect(container.getElementsByClassName('square')[0]).toBeVisible()
    expect(container.getElementsByClassName('line')[0]).toBeVisible()
    expect(container.getElementsByClassName('squareLeft')[0]).toBeVisible()
    expect(container.getElementsByClassName('squareRight')[0]).toBeVisible()

    expect(container.getElementsByClassName('textBlock')[0]).toBeVisible()
    expect(container.getElementsByClassName('blockHeader')[0]).toBeVisible()

    expect(container.getElementsByClassName('title')[0]).toBeVisible()
    expect(getByText('title', { selector: 'div' })).toBeTruthy()

    expect(container.querySelector('h3')).toHaveClass('period')
    expect(getByText('period', { selector: 'h3' })).toBeTruthy()
    
    expect(getByText('children', { selector: 'p' })).toBeTruthy()
  })
})