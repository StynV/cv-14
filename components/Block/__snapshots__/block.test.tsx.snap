// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Block renders a Block 1`] = `
<div>
  <article
    class="block notInView"
  >
    <div
      class="squareBlock"
    >
      <div
        class="square"
      />
      <div
        class="line"
      >
        <div
          class="squareLeft"
        />
        <div
          class="squareRight"
        />
      </div>
    </div>
    <div
      class="textBlock"
    >
      <div
        class="blockHeader"
      >
        <div
          class="title"
        >
          title
        </div>
        <h3
          class="period"
        >
          period
        </h3>
      </div>
      <p>
        children
      </p>
    </div>
  </article>
</div>
`;
